### PR TITLE
feat(utxo-lib): add network configuration for DOGE

### PR DESCRIPTION
**Pull request by DWJ01**
For now we cannot make dogecoin tx with an unspent greater than Number.MAX_SAFE_INTEGER. We will add support for this in the future

BG-51068

### CODE REVIEW

**Review by williamqin-bitgo — CHANGES_REQUESTED** [review]
Need changes to pass unit tests - check out https://github.com/BitGo/BitGoJS/compare/master...BG-49864
(I'm unsure what the dependency audit test fail is though)

**Comment by mmcshinsky-bitgo** [comment]
@DWJ01 Looks like you have lint errors failing your PR.

File: modules/utxo-lib/test/fixtures_thirdparty/nodes/dogecoin-1.14.5/src/test/data/sighash.json [hunk 1]
```diff
@@ -0,0 +1,503 @@
+[
+	["raw_transaction, script, input_index, hashType, signature_hash (result)"],
```
Comment (williamqin-bitgo) [inline modules/utxo-lib/test/fixtures_thirdparty/nodes/dogecoin-1.14.5/src/test/data/sighash.json:2]: out of curiosity, how did you generate this file?

File: modules/utxo-lib/test/fixtures_thirdparty/nodes/dogecoin-1.14.5/src/test/data/tx_valid.json [hunk 2]
```diff
@@ -0,0 +1,514 @@
+[
+["The following are deserialized transactions which are valid."],
```
Comment (williamqin-bitgo) [inline modules/utxo-lib/test/fixtures_thirdparty/nodes/dogecoin-1.14.5/src/test/data/tx_valid.json:2]: same as before, how did you generate this one?

File: modules/utxo-lib/test/networks.ts [hunk 3]
```diff
@@ -41,7 +42,13 @@ describe('networks', function () {
       isLitecoin(network));
 
   const bitcoinTestnetSharedScriptPrefix = (network) =>
-    bitcoinTestnetSharedPubkeyPrefix(network) && !isLitecoin(network);
+    // bitcoinTestnetSharedPubkeyPrefix(network) && !isLitecoin(network);
```
Comment (williamqin-bitgo) [inline modules/utxo-lib/test/networks.ts:45]: I like this, remove the comment though

File: modules/utxo-lib/test/networks.ts [hunk 4]
```diff
@@ -41,7 +42,13 @@ describe('networks', function () {
...
+    isTestnet(network) &&
+    (isBitcoin(network) ||
+      isBitcoinCash(network) ||
+      isBitcoinGold(network) ||
+      isBitcoinSV(network) ||
+      isDogecoin(network));
```
Comment (williamqin-bitgo) [inline modules/utxo-lib/test/networks.ts:51]: while you're at it, maybe make them the same order as `bitcoinTestnetSharedPubkeyPrefix`?

**Review by williamqin-bitgo — DISMISSED** [review]
looks good! squash commits before merging (unless that's done automatically)

File: modules/utxo-lib/test/fixtures_thirdparty/nodes/dogecoin-1.14.5/src/test/data/sighash.json (hunk 1 above)
Comment (DWJ01) [reply modules/utxo-lib/test/fixtures_thirdparty/nodes/dogecoin-1.14.5/src/test/data/sighash.json:2]: It's from [dogecoin github repo](https://github.com/dogecoin/dogecoin/blob/master/src/test/data/sighash.json).

File: modules/utxo-lib/test/fixtures_thirdparty/nodes/dogecoin-1.14.5/src/test/data/tx_valid.json (hunk 2 above)
Comment (DWJ01) [reply modules/utxo-lib/test/fixtures_thirdparty/nodes/dogecoin-1.14.5/src/test/data/tx_valid.json:2]: It's also from [dogecoin github repo](https://github.com/dogecoin/dogecoin/blob/master/src/test/data/tx_valid.json)

File: modules/utxo-lib/test/networks.ts (hunk 3 above)
Comment (DWJ01) [reply modules/utxo-lib/test/networks.ts:45]: removed

File: modules/utxo-lib/test/networks.ts (hunk 4 above)
Comment (DWJ01) [reply modules/utxo-lib/test/networks.ts:51]: changed the order in `bitcoinTestnetSharedPubkeyPrefix` to meet correct alphabetical order.

File: modules/utxo-lib/test/fixtures_thirdparty/nodes/dogecoin-1.14.5/src/test/data/sighash.json (hunk 1 above)
Comment (williamqin-bitgo) [reply modules/utxo-lib/test/fixtures_thirdparty/nodes/dogecoin-1.14.5/src/test/data/sighash.json:2]: !!! I didn't know that's where you find it - awesome!

**Review by williamqin-bitgo — DISMISSED** [review]

File: modules/utxo-lib/test/integration_local_rpc/generate/outputScripts.util.ts [hunk 5]
```diff
@@ -88,7 +88,7 @@ export function createSpendTransactionFromPrevOutputs<T extends UtxoTransaction>
   });
 
   const inputSum = prevOutputs.reduce((sum, { value }) => sum + value, 0);
-  const fee = 1000;
+  const fee = network === utxolib.networks.dogecoinTest ? 1000000 : 1000;
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/test/integration_local_rpc/generate/outputScripts.util.ts:91]:
```suggestion
  const fee = network === utxolib.networks.dogecoinTest ? 1_000_000 : 1_000;
```

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
Please rebase on top of `master` and squash your commits.

Explain the current limits of dogecoin support in the commit message.

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
see comment

**Review by OttoAllmendinger — APPROVED** [review]

**Review by williamqin-bitgo — APPROVED** [review]

**Review by bitgoAaron — APPROVED** [review]

**Pull request merged**